src/core/qgsexpression.cpp

```diff
@@ -4136,7 +4175,13 @@ const QList<QgsExpression::Function *> &QgsExpression::Functions()
                                << Parameter( QStringLiteral( "semi-minor axis" ) )
                                << Parameter( QStringLiteral( "azimuth" ) )
                                << Parameter( QStringLiteral( "segments" ), true, 36 ),
-                               fcnMakeEllipse, QStringLiteral( "GeometryGroup" ) );
+                               fcnMakeEllipse, QStringLiteral( "GeometryGroup" ) )
+        << new StaticFunction( QStringLiteral( "make_regular_polygon" ), ParameterList()
+                               << Parameter( QStringLiteral( "geometry" ) )
+                               << Parameter( QStringLiteral( "geometry" ) )
+                               << Parameter( QStringLiteral( "number of sides/edges" ) )
```

Comment:
missed this above, but these names are variable names, so must be unique for the function, and number_of_sides format only. "semi-minor axis" and the other existing ones will need fixing too.